packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts

```diff
@@ -389,12 +527,63 @@ export class AppSyncSwiftVisitor<
 
   private getInitBody(fields: CodeGenField[]): string {
     let result = fields.map(field => {
-      const fieldName = escapeKeywords(this.getFieldName(field));
-      return indent(`self.${fieldName} = ${fieldName}`);
+      const connectionHasOneOrBelongsTo: boolean = field.connectionInfo
+        ? field.connectionInfo.kind === CodeGenConnectionType.HAS_ONE || field.connectionInfo.kind === CodeGenConnectionType.BELONGS_TO
+        : false;
+      const propertyName = connectionHasOneOrBelongsTo ? `_${this.getFieldName(field)}` : escapeKeywords(this.getFieldName(field));
+      const escapedFieldName = escapeKeywords(this.getFieldName(field));
+      const fieldValue = connectionHasOneOrBelongsTo ? `LazyReference(${escapedFieldName})` : escapedFieldName;
+      return indent(`self.${propertyName} = ${fieldValue}`);
+    });
+
+    return result.join('\n');
+  }
+
+  private getDecoderBody(fields: CodeGenField[]): string {
+    let result: string[] = [];
+    result.push(indent('let values = try decoder.container(keyedBy: CodingKeys.self)'));
+    fields.forEach(field => {
+      const connectionHasOneOrBelongsTo: boolean = field.connectionInfo
+        ? field.connectionInfo.kind === CodeGenConnectionType.HAS_ONE || field.connectionInfo.kind === CodeGenConnectionType.BELONGS_TO
+        : false;
+      const escapedFieldName = escapeKeywords(this.getFieldName(field));
+      const assignedFieldName = connectionHasOneOrBelongsTo ? `_${this.getFieldName(field)}` : escapedFieldName;
+
+      const nativeType = this.getNativeType(field);
+      const optionality = !this.isFieldRequired(field) ? '?' : '';
+      const fieldType = connectionHasOneOrBelongsTo
```

Comment:
The fieldType deduction logic here has 3 levels of ternary conditionals `? : ` making it hard to read, can you please move this to a separate function?